app/js/timer.js

```diff
@@ -1,17 +1,34 @@
-/**
- *  sets a 30 minute timer for taking the test
+/** function tmeElapsed takes a dateStamp and return the difference between the current dateStamp and
+ *  the one that is passed into the function
+ *
+ * @param dateStamp parameter is the date stamp that is stored in a cookie at the beginning of the test
+ * @return returns the difference between the date stamp of the moment the function is called and the
+ * parameters date stamp value
  */
+function timeElapsed(dateStamp){
+    let timeNow = Date.now()
+    return  timeNow - dateStamp
+}
 
-//30 minute time limit
-let timeLimit = 1799
-
+/**
+ *Function timer() figures out how many seconds and minutes remain and displays this value. When called 1 is
+ * taken off the value of timeRemaining.
+ * If the timeRemaining is less than 0 then the function (not our code) automatically clicks on the
+ * finish button.
+ */
 function timer() {
-    let minutes = Math.floor(timeLimit / 60);
-    let seconds = Math.floor(timeLimit - minutes * 60);
+    //30 minute time limit in seconds (plus one second for lag).
+    const timeLimit = 1801
```

Comment:
by putting this in here instead of outside of the function, you are redeclaring the constant once every second. this is unnecessary processing and technically throws a JS error, except u dont see it because const is a property of an object by that point so can be redeclared. still not a good thing. move this out of the function